src/Axe/Views/Compiler/Solve.cshtml

```diff
@@ -1,87 +1,102 @@
 @using Axe.Models;
 @model Axe.ViewModels.CompilerVm.CodeBlockSolveVm
+
 @{
     ViewData["Title"] = "Solve task #" + Model.Id;
 }
 
 @section Scripts
 {
     <script src="~/js/compiler.solve.js"></script>
+    @if (Model.isMonitoringMode && ViewData["observedUserId"] != null)
+    {
+        <script src="~/js/compiler.solve.observer.js"></script>
+    }
+    else
+    {
+        <script src="~/js/compiler.solve.user.js"></script>
+    }
+}
+
+@if (Model.isMonitoringMode && ViewData["observedUserId"] != null)
+{
+    <input id="observedUserId" value="@ViewData["observedUserId"]" hidden="hidden" />
 }
 
+<input id="taskId" asp-for="Id" value="@Model.Id" hidden />
+<input asp-for="Task" value="@Model.Task" hidden />
+
 <div class="row" style="margin-top: 1em;">
-    <div class="col-md-12">
-        <div class="panel panel-default">
-            <div class="panel-heading"><b>Task:</b> @Model.Id</div>
-            <div class="panel-body">@Model.Task</div>
+    <div class="col-md-4">
```

Comment:
area for task description is too small. i suggest to use entire "row" and also allow markdown in task description to highlight code, bold, etc